### ISSUE TITLE
Multiple Types of Currency

### ISSUE THREAD

**Issue by TyHammer125** [HabitRPG/habitica#438]
I think one good idea could be to have a couple different types of currency. Then you could force yourself to have certain rewards for only doing certain tasks.

For example, if you are on a diet, you need to unlock a currency like sapphires to eat junk food. You can only unlock sapphires from working out though.

**Comment by USM-Valor** [comment]
Great idea.

**Comment by wc8** [comment]
I like the idea also, had been wondering about some way of linking rewards to types of tasks.
[Tags](https://trello.com/c/q7v02jAw) is on the feature request list. Perhaps a kind of gemstone (or color of gemstone) could be attached to each tag. You would then gain gemstones based on habits/daily tasks/todos done for that tag. Rewards could be assigned Gold and/or a gemstone type(s). 

Possibilities:
- You set _Health_ tag to yield red gemstone (rubies), _Study_ tag to yield blue gemstone (sapphires), in addition to regular GP.
- _workout_ and _eat healthy lunch tagged health. You assign *eat cake_ 1 gemstones and 0 GP. You can always only _eat cake_ if you either worked out or ate a healthy lunch. 
- 1 hour of study history and 1 hour of read paper are tagged study. You assign reward _Watch TV episode_ the cost of 1 Ruby, 1 Sapphire and 20 GP. That means you have to work out or eat a healthy lunch **and** do some study before you can watch TV.

Also, this would maybe be a way to provide rewards tied directly to specific completion without changing the GP algorithm.

[Updated description]

**Comment by toebu** [comment]
I like this idea too. This guy from lifehacker for sure would also like it: http://lifehacker.com/5975824/gamify-your-life-a-guide-to-incentivizing-everything
You're right @wc8, this might indeed help with my #392. Other people commented on that suggestion, that it might get too complicated if we add another type of money. I agree with that. I also think that introducing different types of money and gems would mess with the algorithm that determines the value of a task. Would you always get 1 Ruby for completing '1 hour of study' even though the GP/EXP amount will decrease? That might be confusing.
So here is an idea combining the different types of money from here with my idea over on #392 to introduce a money chest:
- There are different types of pots (or purses or chests). 
- For every task, you can define how much of the GP goes into which pot (as a percentage, split evenly between the pots you select or with tags somehow)
- For every reward, you can define how much it costs from which pot

So you could have pots for e.g. health or study. And of course you could have a pot for long-term rewards (going for a trip, buying a new gadget, ...) and in-game rewards (armor, weapons, ...).

The money in one pot doesn't have to necessarily stay in that pot. I would say that one could either make transactions once a week, or just move the money around freely. This feature would mostly be to keep track of what one has earned for which type of activity, and not to stop people to buy what they want, if they really want it.

**Comment by wildcate** [comment]
That, to me, sounds like a good solution - no need to make up a pot if one is okay with just one sort of currency, and endless possibilities to spread the rewards into different pots if inclined to do so. Also - no algorithm troubles!

**Comment by PhilipHow** [comment]
I do like the pot idea, but I'm not sure how it translates well on the page. Any ideas?

**Comment by wc8** [comment]
@toebu that sounds like the envelope budget system (which I find appealing). The only thing missing would be the link between categories of tasks and possible rewards. With the pot system, can one require "take the stairs" type habits/tasks for the "eat cake" reward?

As far as complexities for the user, gemstone use would optional, users could stick with the familiar, simple GP and ignore gemstones if desired. Also, gemstones could be introduced (unlocked) later. 
Actually, I prefer a term like "gemstones" rather than the term multiple currencies, because the latter seems daunting. _What, I have to calculate in euros, dollars and yen now? What's the rate of exchange?_ when there is no rate of exchange to think about.

As far as the algorithm goes, I don't see how gemstones as a fixed points would mess with the algorithm. It would be independent of it. GP with its algorithms would be the currency. Gemstones would just give the user a chance to make certain rewards constants. 
For example: 
_Taking the stairs_, is always rewarded with a chance to _eat a cookie_ **or** _have some candy_, plus algorithm-based GP towards bigger rewards like watching TV, buying leather armor, etc.. As you gain mastery of the stairs habit, the value in terms of general rewards lessens, but the value in terms of caloric intake stays the same.

**Comment by toebu** [comment]
Never heard of the envelope budget system before, but it sounds pretty much like it. The link between categories of tasks and rewards is there: for your "take the stairs" habit, you simply say that all the money goes into the "sweets" pot. The "eat cake" reward would then be configured to cost money from that pot. (One could use tags for that, e.g. the "pot-sweets" tag puts all the GP in that pot.)

Of course anything we're thinking of here should be optional. Both the gemstones and the pots. And both could be unlocked later, so there's no difference there between the two ideas.

Why I'm rather against the gemstones: This adds another layer of complexity to setting the cost of a reward. After playing a while, you will notice how much GPs you get per day/week on average. Based on this you can then make your reward costs. With gemstones, this will be more difficult. In your example, I wonder how you decide that "Watch TV episode" costs 1 Ruby, 1 Sapphire and 20 GP. I imagine it would be quite difficult to find out which tasks you exactly have to complete to get that reward.
One advantage of the gemstones would indeed be that you can set a constant value for a task. But I don't understand why you would want the GP to vary, but the gemstones to stay constant for any particular task. For me it makes more sense to either say, this is a task that I always want to get the same for, or you want it to vary depending how well your doing. Hence my suggestion over on Trello ([Priority Multiplier](https://trello.com/card/priority-multiplier/50e5d3684fe3a7266b0036d6/17)), that one should be able to set for every task whether the value should change or not.
Don't get me wrong @wc8, I think the gemstones would also work and achieve more or less the same outcome. But I simply think it would be a bit more complicated to work with.

As for @MrConcept how this would fit in the page. Over on #361 there are some mockups for how it might look like soon. I'm sure one could fit this in the header somewhere, or maybe only put the details of how much GP is in which pot in the 'Ye Olde Town Shop' Modal, where we'll have more space.

**Comment by TyHammer125** [comment]
I think either would work, but I feel that the pot system would be easier to cheat for users. If someone gets lazy, you could just allocate all the gold you have to the "sweets pot" with the number of transactions you have that week. Whereas with gemstones, you wouldn't be able to create them. 

@toebu, as for how to determine what combination of gemstones should a reward be worth, I believe users would make rewards proportional to the gemstones of the tasks they need done. For example, a student would probably make watching TV proportional to the gemstones that correspond to studying and working out.

I agree with @wc8 that gemstones would not mess with the algorithm. 
What I was thinking of the implementation was when you set a task, there is a box you can check for rewarded gemstones. Once you check it, it shows the gemstones you have unlocked and you can allocate a number of gemstones as a reward for that task. Then when you complete the task, the algorithm rewards you the gold and experience you would normally get for that task plus the gemstones you allocated as a bonus.

For example, you could have two tasks and two rewards. Taking the stairs rewards you what the algorithm normally rewards you plus 1 emerald. Running rewards you what the algorithm normally rewards you plus 3 emeralds. Watching an hour of TV would cost 20 GP plus 1 emerald while eating cake costs 20 GP plus 3 emeralds.  

Also as a last note, @wc8, I definitely like calling this gemstones over multiple types of currency.

**Comment by toebu** [comment]
There's always a way to cheat. We really shouldn't worry too much about that. With the gemstones, if you really want to have the sweet, even though you didn't deserve it you still have loads of way to get it anyway:
- Just eat the sweet and ignore HabitRPG
- Change the cost of the reward to not include the gem you don't have
- Make a task, assign it the missing gem and mark it done
  Actually, we would even lose one feature that will stop you from cheating: at the moment, if you buy a reward for which you don't have enough GP, you lose HP for the missing GP. So if you have 15 GP and buy something that costs 20 (except of course in-game items), you are down on 0 GP and lose 5 HP (= 20 - 15). With the pots, that mechanism would still work. What would happen in this situation with a missing gem? You are still going to eat that candy if you're really set on it. So you have to cheat in one of the ways described above. Or of course we say that for every gem you don't have, you lose e.g. 5 HP. But then we're again at the point where there is a sort of exchange rate between GP and gems, which there shouldn't be.

I also still think that figuring out a cost in GP and various gems is more difficult than only GP. In your example @TyHammer125, I understand that eating cakes costs 3 emeralds because you want to get it for either going running once or taking the stairs 3 times. But where does the 20 GP cost come from?

Another thing we would lose with the gems is granularity: the costs of my rewards are quite fine tuned (using e.g. 1.5 GP), we probably don't want to break the gemstones, so I could only set whole numbers for costs.

Finally, if you are still concerned with cheating, one could also say that you cannot move money from one pot to another at all (but you could still cheat by changing the reward cost to use money from another pot...).

**Comment by wildcate** [comment]
For myself, I would not want gemstones. To me, it seems much too complicated, and I agree that figuring out the rates for gemstones in addition to gold would be extra-hard. And I found it hard enough to set the rewards when I started out... how much gold will I have to spend for cake, or for buying a book as a treat? I don't think I would have started with HRPG if there had been multiple currencies from the start.

And if you add the gemstone possibility later... that will mess up (or mean re-defining) all of the set rewards that you already have. Whereas if you just pot the money, it won't necessarily change the overall value, just what you have to perform as habit to earn that money.

So basically, I totally agree with what @toebu said here. If it's about a way to keep track of whether you ate healthy enough to eat some candy now, the pots would work fine in my opinion, without adding extra complicatedness.

**Comment by wc8** [comment]
I wasn't worried/thinking about cheating at all. I was just thinking of keeping track. 
If the pots could be tagged or somehow linked to types of tasks, that's fine enough for me. Maybe I missed that in the suggestion. The key, for me, is a link between categories of tasks and categories of rewards.

**Comment by wc8** [comment]
In other words, I would want the completion of a particular task category to automatically contribute to a particular pot and rewards to be extracted from particular pot(s).

As far as the ability to move money between pots, with the way the game works, it's probably going to be hard enough to reach that setting that users won't be too encouraged to do it. Furthermore I can see a use for allowing it. What happens if, after 14 flights of stairs to the doctor's office, you get home and find you're all out of cookies and cake? No reward? Better to allow some redistribution of funds.

**Comment by toebu** [comment]
Should I add this to trello then, what do you think?

**Comment by wc8** [comment]
Sure.

**Comment by toebu** [comment]
Finally created a [trello card](https://trello.com/card/split-earned-money-into-different-pots/50e5d3684fe3a7266b0036d6/224). Make me happy and go vote for it :)
Closing this ticket, can be re-opened if/when this will be implemented.
